ulib/experimental/Steel.Memory.fst

```diff
@@ -75,14 +112,23 @@ let disjoint_sym (m0 m1:heap)
 
 let join (m0:heap) (m1:heap{disjoint m0 m1})
   : heap
-  = F.on _ (fun a ->
+  = on _ (fun a ->
       match m0 a, m1 a with
       | None, None -> None
       | None, Some x -> Some x
       | Some x, None -> Some x
       | Some (Ref a0 p0 v0), Some (Ref a1 p1 v1) ->
-        Some (Ref a0 (sum_permissions p0 p1) v0))
+        Some (Ref a0 (sum_permissions p0 p1) v0)
+      | Some (Array a0 len0 seq0), Some (Array a1 len1 seq1) ->
+        Some (Array a0 len0 (Seq.init len0 (fun i ->
+	  let (x0, p0) = Seq.index seq0 i in
+          let (_, p1) = Seq.index seq1 i in
+	  (x0, (sum_permissions p0 p1 <: (perm:permission{allows_read perm})))
```

Comment:
This would also have to change to account for array splitting.

Reply:
Indeed, it has been changed